src/apps/Permissions/Home/AllPermissions.js

```diff
@@ -0,0 +1,189 @@
+          )
+        })
+
+        // Filter by account types
+        .filter(permission => {
+          // accounts
+          if (selectedEntityType === 1) {
+            return permission.entities.some(entity => entity.type !== 'app')
+          }
+
+          // apps
+          if (selectedEntityType === 2) {
+            return permission.entities.some(entity => entity.type === 'app')
+          }
+
+          return true
+        })
+    )
+  }, [searchTerms, permissions, selectedEntityType])
+
+  if (loading) {
+    return <EmptyBlock>Loading permissions…</EmptyBlock>
+  }
+
+  if (permissions.length === 0) {
+    return <EmptyBlock>No permissions have been assigned yet.</EmptyBlock>
+  }
+
+  return (
+    <PermissionsView
+      permissions={filteredPermissions}
+      onManageRole={onManageRole}
+      heading={
+        layoutName === 'large' && (
+          <Heading
+            emptyFilter={Boolean(
+              !filteredPermissions.length && permissions.length
+            )}
+            selectedEntityType={selectedEntityType}
+            onEntityTypeChange={setSelectedEntityType}
+            searchTerms={searchTerms}
+            onSearchTermsChange={setSearchTerms}
+          />
+        )
+      }
+      showApps
+    />
+  )
+}
+
+AllPermissions.propTypes = {
+  loading: PropTypes.bool.isRequired,
+  onManageRole: PropTypes.func.isRequired,
+  permissions: PropTypes.array.isRequired,
+}
+
+function Heading({
+  emptyFilter,
+  selectedEntityType,
+  onEntityTypeChange,
+  searchTerms,
+  onSearchTermsChange,
+}) {
+  const theme = useTheme()
+
+  const handleEntityDropDownChange = useCallback(
+    index => {
+      onEntityTypeChange(index || -1)
+    },
+    [onEntityTypeChange]
+  )
+  const handleSearchInputChange = useCallback(
+    event => {
+      onSearchTermsChange(event.target.value)
+    },
+    [onSearchTermsChange]
+  )
+  const handleOnClear = useCallback(() => {
+    onEntityTypeChange(-1)
+    onSearchTermsChange('')
+  }, [onEntityTypeChange, onSearchTermsChange])
+
+  return (
+    <React.Fragment>
+      <div
+        css={`
+          display: flex;
+          width: 100%;
+          justify-content: space-between;
+          align-items: center;
+        `}
+      >
+        <div
+          css={`
+            white-space: nowrap;
+            ${textStyle('body2')}
+            font-weight: 600;
+          `}
+        >
+          All assigned permissions
+        </div>
+        <div>
+          <label
+            css={`
+              margin-right: ${2 * GU}px;
+              color: ${theme.surfaceContentSecondary};
+            `}
+          >
+            <DropDown
+              placeholder="Entity type"
+              items={ENTITY_TYPES}
+              selected={selectedEntityType}
+              onChange={handleEntityDropDownChange}
+              css={`
+                min-width: ${16 * GU}px;
+              `}
+            />
+          </label>
+          <TextInput
```

Comment:
I found this odd readjustment of items when searching, maybe there's nothing we can do right now:
https://www.loom.com/share/3d168f03a3554611a928aa41a6d73af0

Reply:
Yes `DataView` uses the table layout, which just adapts to its content. It’s a difficult one to fix because we really don’t know what is going to be in every cell.

But maybe we could have a way to provide column widths? So that it can optionally be fixed, when having a live filtering feature.

Other than that, we should debounce this! :smile:

Reply:
Done https://github.com/aragon/aragon/pull/899/commits/6d7ac94403581e17b2e5a2c50bc815531e200fb5